### ISSUE TITLE
build nginx source 1.15.1 with fastdfs-nginx-module error

### ISSUE THREAD

**Issue by neocxf** [happyfish100/fastdfs-nginx-module#28]
I check out the source code of nginx with version 1.15.1, and the file structure is like this:
![ngix-src-structure](https://user-images.githubusercontent.com/12878595/42617798-79da4764-85e5-11e8-8a36-3bd576bb0b57.png).

So, first the INSTALL doc should change a little bit. When I enter nginx release folder, It seems that I should execute something like this `auto/configure --add-module` to start the configure

Secondly, even if I finish the first process. I execute the `make` command. It gives me this error:
![make-error](https://user-images.githubusercontent.com/12878595/42617891-e5d81e28-85e5-11e8-8522-67fb35540402.png)

Basically, It complains about the existence of `common_define.h`. But I have already installed the [libfastcommon ](https://github.com/happyfish100/libfastcommon)library. And I can find the header file inside the `/usr/include/fastcommon` folder.

I have already tried manually adding the -I option to the nginx makefile. But after the compilation and installation, when I start the nginx process, the fastdfs-nginx-module is still not working.

Any help? please

**Comment by happyfish100** [comment]
you should use the last version: fastdfs-nginx-module V1.20, FastDFS V5.12 and libfastcommon V1.0.38.

**Comment by neocxf** [comment]
I have already try to rebuild from the source that I just pull from github. The whole make process seems OK. and there is not any error when make the nginx source.

But, when I start the process, and make a request to the server, it's not working. The nginx error.log give this:

![image](https://user-images.githubusercontent.com/12878595/42749834-a8f43fe8-8917-11e8-9f2b-a6ab19521df6.png)

It seems that there are still something wrong about the including header.

**Comment by happyfish100** [comment]
http.conf  is in FastDFS source, the file path is conf/http.conf.
you should copy it to /etc/fdfs/http.conf.

**Comment by happyfish100** [comment]
you should change the file permissions to the user and group who really run nginx, such as nginx:nginx. or you can set permission 755 to /root/fdfs/file and it's sub-dirs.

**Comment by neocxf** [comment]
Thanks, I finally get everything done. It turns out that by default the nginx will spawn a child worker process by `nobody`. In this case, if I tried to access the file, I got file permission issues.
![image](https://user-images.githubusercontent.com/12878595/42792109-648e53cc-89a6-11e8-83e5-57eced83b284.png)

So the correct way I think is to make sure that the worker process and the requesting file share the same user and group. If they are not same, you should change your `nginx.conf` file, alter the user. After that, restart the nginx, then everything is done.
